cmd/argocd/commands/app.go

```diff
@@ -1733,9 +1784,18 @@ func waitOnApplicationStatus(acdClient argocdclient.Client, appName string, time
 			}
 		}
 		_ = w.Flush()
+		if opts.watch.delete {
+			_, err := appClient.Get(context.Background(), &applicationpkg.ApplicationQuery{Name: &opts.appName})
+			if err != nil {
+				if strings.HasSuffix(err.Error(), "not found") {
```

Comment:
```suggestion
				if apierr.IsNotFound(err) {
```

Maybe using `IsNotFound(err)` error type check from "k8s.io/apimachinery/pkg/api/errors" is a better option here?